### DISCUSSION TITLE
Netbox 2.11.11 region nesting errors: "dcim_region" violates check constraint "dcim_region_rght_check"

### DISCUSSION THREAD

**Discussion by kevclarx** [netbox-community/netbox#7345] [Q&A]
We recently decided to change how our netbox regions were laid out and began seeing very strange behavior in the web ui.  

Site counts were incorrect on the region view for some regions, until you click into them and they show the correct sites.  Regions suddenly showed up under other regions even though their parent region is correct when you click into that region, but the breadcrumb shows all kinds of wacky hierarchy.

I've tried invalidating cache and that did not seem to do anything, and for some regions if I try to reassign their parent I now get:
```
<class 'django.db.utils.IntegrityError'>
new row for relation "dcim_region" violates check constraint "dcim_region_rght_check"
```

Any ideas?

**Comment by DanSheps** [comment]
What version of netbox and has anyone edited anything manually in the database?

**Reply by kevclarx** [reply]
2.11.11, no one edited anything manually in the db this was caused by a user moving regions around in the web ui only.

**Reply by kevclarx** [reply]
I'm browsing the db with pgadmin and here is what it looks like(region names scrubbed)
![dcim_regions_tree](https://user-images.githubusercontent.com/7095848/134421479-56d0c4e8-5cd7-4201-b66b-e14d11017ca8.png)

**Reply by kevclarx** [reply]
I was able to correct this with a little reading on MPTT(https://django-mptt.readthedocs.io/en/latest/overview.html) and manually setting the correct lft, rght, tree_id, and level fields.  I'm not sure if I will be able to reproduce how we got into the bad state but will play with it a bit to see if I can replicate.
